pkg/storage/image.go

```diff
@@ -57,13 +57,15 @@ type imageCacheItem struct {
 	configDigest digest.Digest
 }
 
+type imageCache map[string]imageCacheItem
+
 type imageService struct {
 	store                 storage.Store
 	defaultTransport      string
 	insecureRegistryCIDRs []*net.IPNet
 	indexConfigs          map[string]*indexInfo
 	registries            []string
-	imageCache            map[string]imageCacheItem
+	imageCache            imageCache
```

Comment:
is cache necessary at all? lets remove it if possible :) nice sweet negative patch that would be

Reply:
We're trying to save the effort of JSON umarshalling that `NewImage()` does, so we cache the things about the image that don't (can't) change and that aren't directly in the `storage.Image` that we're already looking up, which we would have to parse the image's manifest and/or config blob to figure out.  On a node with a large number of images, that's a lot of parsing data that doesn't change over the lifetime of the image, to obtain the same results every time.

Reply:
So there no remote calls made at all? Is there any bechmark to check to see the effect of caching?  I don't have up to date numbers, but standard encode/json was able to do ~30000 unmarshals/second/core for ~4KB json.  Random link from the internet tells similar story: https://github.com/json-iterator/go-benchmark 

How many images you are aiming for?

Reply:
Decoding the manifest requires guesswork because we don't know the type of the manifest when reading it from disk, and we also have to read and decode the config blob, if there is one, and we get the results are the same every time.  When we started parsing them to get user information in #1332, I suggested caching it to avoid taking a speed hit, but it wasn't measured beyond an anecdotal check with `crictl images` with a dozen or two images.  If benchmarks aren't usefully affected, I'm fine with dropping it.

The kubelet's garbage collector appears to be calling `ListImages` with no filter every minute or so (the code suggests 30 seconds to me, but the docs say every minute, so I must be reading it wrong), and I tend to assume that we'll have around a thousand images at the outside because people do unexpected things.

Reply:
Does any of e2e tests you have runs with >1k images so that we can try to remove it and compare run times?